content/docs/08-troubleshooting/01-migration-guide/04-migration-guide-4-0.mdx

```diff
@@ -245,6 +245,16 @@ Please use the `streamProtocol` parameter instead.
 The `experimental_useAssistant` export has been removed from the `useAssistant` hook.
 Please use the `useAssistant` hook directly instead.
 
+#### Removed `threadId` and `messageId` from `AssistantResponse`
+
+The `threadId` and `messageId` parameters have been removed from the `AssistantResponse` function.
+Please use the `threadId` and `messageId` variables from the outer scope instead.
```

Comment:
from the outer scope isn't super clear to me. However, I haven't done anything with Assistant API so might be obvious.